specification/nginx/resource-manager/NGINX.NGINXPLUS/preview/2025-03-01-preview/swagger.json

```diff
@@ -0,0 +1,2850 @@
+            "description": "The policy is referenced in the nginx config and is applying."
+          },
+          {
+            "value": "Succeeded",
+            "description": "The policy is referenced in the nginx config and that config has been successfully applied."
+          },
+          {
+            "value": "Failed",
+            "description": "The policy is referenced in the nginx config and that config failed to apply."
+          },
+          {
+            "value": "Removing",
+            "description": "The policy is now not referenced in the nginx config and its being removed from the applied nginx config."
+          }
+        ]
+      }
+    },
+    "NginxDeploymentWafPolicyCompilingStatus": {
+      "type": "object",
+      "title": "Nginx Deployment Waf Policy Compiling Status",
+      "readOnly": true,
+      "properties": {
+        "code": {
+          "$ref": "#/definitions/NginxDeploymentWafPolicyCompilingStatusCode"
+        },
+        "displayStatus": {
+          "type": "string",
+          "readOnly": true,
+          "description": "A readable string of the current status, and sometimes have the reason for the current state. If the CompilingStatus is Failed the Display Status will be The waf Policy failed to compile."
+        },
+        "time": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The date and time the policy was compiled in UTC."
+        }
+      }
+    },
+    "NginxDeploymentWafPolicyCompilingStatusCode": {
+      "type": "string",
+      "example": "InProgress",
+      "readOnly": true,
+      "description": "Machine readable code indicating the compilation status of a WAF Policy.",
+      "enum": [
+        "NotStarted",
+        "InProgress",
+        "Succeeded",
+        "Failed"
+      ],
+      "x-ms-enum": {
+        "modelAsString": true,
+        "name": "NginxDeploymentWafPolicyCompilingStatusCode",
+        "values": [
+          {
+            "value": "NotStarted",
+            "description": "The compilation of the custom waf policy has not started"
+          },
+          {
+            "value": "InProgress",
+            "description": "The compilation of the custom waf policy is in progress"
+          },
+          {
+            "value": "Succeeded",
+            "description": "The compilation of the custom waf policy is completed successfully and can now be referenced in the nginx config."
+          },
+          {
+            "value": "Failed",
+            "description": "The compilation of the custom waf policy failed."
+          }
+        ]
+      }
+    },
+    "NginxDeploymentDefaultWafPolicyListResponse": {
+      "type": "object",
+      "title": "Nginx Deployment Default Waf Policy List Response",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/NginxDeploymentDefaultWafPolicyProperties"
+          },
+          "x-ms-identifiers": []
```

Comment:
wouldn't they usually have an identifying property? such as the 'id'?

Reply:
No we don't have any uniquely identifying properties apartment from the Azure Resource ID